BioCro/DARPA/upload_params.Rmd

```diff
@@ -177,3 +177,55 @@ Save cuticular conductance values as a .csv.
 write.csv(upload_cc, file = "upload_params_data/cut_cond.csv", row.names = FALSE)
 ```
 
+## Greenhouse physiological parameters Vmax, Rd, and stomatal slope
+
+Same as physiological parameters as above, but for greenhouse experiments
+
+Clean up data to have the final columns: 
+
+1. `notes`: records which row in the Google Sheet the record came from
+2. `local_datetime`: convert date into machine readable format
+3. `treatment`: specifies record's treatment using BETYdb treatment names
+4. `Vcmax`, `leaf_respiration_rate_m2`, `stomatal_slope.g1`, `stomatal_slope.BB`: measured values of physiological parameters specified
+5. `leafT`: leaf temperature, which is required by BETYdb
+
+```{r}
+upload_parameters_gr <- all_parameters %>% 
+  mutate(notes = paste("row", row_number(), "in https://docs.google.com/spreadsheets/d/1doQI0T7vav7LmNdMEoZJDx9iY_Imqa229CLRbRvurFc/#gid=967233489")) %>% 
+  filter(genotype == "ME034", 
+         TR %in% c("greenhouse"), 
+         trait %in% c("vmax", "Rd", "g1M", "g1BB")) %>% 
+  mutate(value = ifelse(!is.na(New_Value), New_Value, Value), 
+         se = ifelse(!is.na(New_Value), SE.1, SE)) %>% 
+  mutate(date = ymd(paste0(str_sub(File, -4, -1), 
+                           "-", match(str_sub(File, 1, 3), month.abb), 
+                           "-", str_sub(File, -6, -5))), 
+         treatment = case_when(TR == "greenhouse" ~ "greenhouse")) %>% 
+  select(-File, -genotype, -TR, -rep, -method, -Value, -SE, -New_Value, -SE.1, -SD) %>% 
+  spread(trait, value) %>% 
+  rename(local_datetime = date, 
+         Vcmax = vmax, 
+         leaf_respiration_rate_m2 = Rd, 
+         stomatal_slope.g1 = g1M, 
+         stomatal_slope.BB = g1BB, 
+         SE = se) %>% 
+  mutate(leafT = 25, 
+         n = 9999) %>% 
+  replace(is.na(.), "")
+```
+
+Save cleaned parameter as a .csv file in `upload_params_data` folder in `model-vignettes/BioCro/DARPA`. 
+```{r}
+if(!dir.exists("upload_params_data/")){
+  dir.create("upload_params_data")
+}
+
+write.csv(upload_parameters_gr, "upload_params_data/phys_parameters_gr.csv", row.names = FALSE)
+```
+
+Upload that new parameter data file using the [Bulk Upload](http://welsch.cyverse.org:8000/bety/bulk_upload/start_upload) in Welsch BETYdb. The following additional data, which applies across all the parameters, will have to be entered
```

Comment:
We're going to have to create a "greenhouse" treatment in the database before uploading too.

Reply:
This will actually probably need to be at its own site in the database.